src/client/MainFrameController.java

```diff
@@ -17,9 +20,15 @@
     private int MAX_LOGIN_ATTEMPTS = 3;
     private int loginAttempts;
     private CustomerService customerService;
+    private AddressService addressService;
+    private CreditCardService creditCardService;
+    private Session session;
 
-    public MainFrameController(CustomerService customerService) {
+    public MainFrameController(Session session, CustomerService customerService, AddressService addressService,
```

Comment:
I think it would be better if we abstracted the Services with to an interface Service, then added a method to the MainFrameController called `addServices(ArrayList<Service> services)` to add in the services. This is more of a design thing and doesn't have to be done if we don't have the time.